src/libs/Navigation/AppNavigator/MainDrawerNavigator.js

```diff
@@ -91,6 +94,11 @@ class MainDrawerNavigator extends Component {
             return false;
         }
 
+        if (!this.initialParams.reportID) {
+            const state = this.props.navigation.getState();
+            const reportScreenKey = lodashGet(state, 'routes[0].state.routes[0].key', '');
+            Navigation.setParams(initialNextParams, reportScreenKey);
```

Comment:
Why is this needed?

Reply:
To update the `initialParams` of `ReportScreen`. When we load the data for the first time, the initial `initialParams` will have an empty `reportID`, so we need to update it later after we get the `reportID`. A screen that is already mounted need to use `setParams` to update the `initialParams`.

https://reactnavigation.org/docs/navigation-actions/#setparams

Reply:
Should I add this comment?

`// Update the report screen initial params after the reports are available`

Reply:
```
// Update the report screen initial params after the reports are available
// to show the correct report instead of the "no access" report.
// https://github.com/Expensify/App/issues/12698#issuecomment-1352632883
```

Do you think this is good enough?

Reply:
Ok, I think I get it now. Thanks